File: gestion_cinema/src/main/java/net/yacine/gestion_cinema/entities/Film.java
package net.yacine.gestion_cinema.entities;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Builder;

@Entity @NoArgsConstructor @AllArgsConstructor @Data @Builder
public class Film {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String description;
    private String director;
    private LocalDate releaseDate;
    private LocalTime duration;
    private String image;
    @ManyToOne
    private Category category;
    @OneToMany(mappedBy = "film")
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private List<FilmScreening> filmScreenings;
}

File: front_end_gestion_cinema/lib/data/models/film.dart
import 'package:front_end_gestion_cinema/data/models/category.dart';

class Film {
  late String title;
  //late String description;
  //late String director;
  //late String releaseDate;
  late String duration;
  late String image;
  late String urlCategory;
  Category category = Category();

  Film(){}

  Film.fromJson(Map<String, dynamic> json) {
    title = json['title'];
    //description = json['description'];
    //director = json['director'];
    //releaseDate = json['releaseDate'];
    duration = json['duration'];
    image = json['image'];
    urlCategory = json["_links"]["category"]["href"];
  }
}
